### DISCUSSION TITLE
How to define Layout in a <script setup> tag?

### DISCUSSION THREAD

**Discussion by Beethoven** [nuxt/nuxt#18286] [Questions] [answered]
How to define Layout in a <script setup> tag?
May be an example in docs?
https://v3.nuxtjs.org/docs/directory-structure/layouts

**Comment by didavid61202** [comment, marked as answer]
same as #882, make sure to check issues and discussions before creating new one.

you can have both **normal script** tag and **setup script** tag in same SFC, 
so just add another script tag like:

```ts
<script lang="ts">
export default {
  layout: "custom",
};
</script>
```

note: sure both setup script and normal script tag have lang="ts"

**Reply by Beethoven** [reply]
I saw the  #882.
Sorry for the duplicate.

What's the difference from

`<script lang="ts">
export default {
  layout: "custom",
};
</script>`

to 

`<script lang="ts">
export default defineNuxtComponent({
  layout: "custom",
});
</script>`

**Reply by wobsoriano** [reply]
type inference

**Comment by sduduzog** [comment]
Sorry to throw this question here but, do you guys have volar issues with typescript properties withing the vue <template> ?

I know there's about 2 seperate discussions about it somewhere but none are clear if anyone actually has that working with the @nuxt/bridge or not.

**Comment by aaronransley** [comment]
I have created a PR for this clarification: #1305 

@wobsoriano Do you think the example in the PR should [use a `lang="ts"` example](https://github.com/nuxt/framework/discussions/1276#discussioncomment-1508371), or stick with vanilla JS?

The other examples [in the layouts page](https://v3.nuxtjs.org/docs/directory-structure/layouts) are using JS, and I'm not sure if there's a standard re: the docs consistently using either JS or TS, or if examples should be provided for both.

**Reply by wobsoriano** [reply]
> I have created a PR for this clarification: #1305 
> 
> @wobsoriano Do you think the example in the PR should [use a `lang="ts"` example](https://github.com/nuxt/framework/discussions/1276#discussioncomment-1508371), or stick with vanilla JS?
> 
> The other examples [in the layouts page](https://v3.nuxtjs.org/docs/directory-structure/layouts) are using JS, and I'm not sure if there's a standard re: the docs consistently using either JS or TS, or if examples should be provided for both.

Js should be fine 👌
